plugin-server/tests/worker/ingestion/cookielessServerHashStep.test.ts

```diff
@@ -0,0 +1,371 @@
+                    $ip: '5.6.7.8',
+                },
+                uuid: new UUID7(now.getTime()).toString(),
+            })
+            eventDifferentDay = deepFreeze({
+                ...event,
+                now: differentDay.toISOString(),
+                uuid: new UUID7(differentDay.getTime()).toString(),
+            })
+            identifyEvent = deepFreeze({
+                event: '$identify',
+                distinct_id: identifiedDistinctId,
+                properties: {
+                    [COOKIELESS_MODE_FLAG_PROPERTY]: true,
+                    $anon_distinct_id: COOKIELESS_SENTINEL_VALUE,
+                    $host: 'https://example.com',
+                    $raw_user_agent: userAgent,
+                    $ip: '1.2.3.4',
+                },
+                site_url: 'https://example.com',
+                team_id: teamId,
+                now: now.toISOString(),
+                uuid: new UUID7(now.getTime()).toString(),
+                ip: null,
+            })
+            identifyEventABitLater = deepFreeze({
+                ...identifyEvent,
+                now: aBitLater.toISOString(),
+                uuid: new UUID7(aBitLater.getTime()).toString(),
+            })
+            postIdentifyEvent = deepFreeze({
+                event: 'test event',
+                distinct_id: identifiedDistinctId,
+                properties: {
+                    [COOKIELESS_MODE_FLAG_PROPERTY]: true,
+                    $host: 'https://example.com',
+                    $raw_user_agent: userAgent,
+                    $ip: '1.2.3.4',
+                },
+                site_url: 'https://example.com',
+                team_id: teamId,
+                now: now.toISOString(),
+                uuid: new UUID7(now.getTime()).toString(),
+                ip: null,
+            })
+            aliasEvent = deepFreeze({
+                ...event,
+                event: '$create_alias',
+            })
+            mergeDangerouslyEvent = deepFreeze({
+                ...event,
+                event: '$merge_dangerously',
+            })
+            nonCookielessEvent = deepFreeze({
+                ...event,
+                properties: {
+                    $host: 'https://example.com',
+                    $raw_user_agent: userAgent,
+                },
+            })
+        })
+
+        // tests that are shared between both modes
+        describe.each([
+            ['stateless', CookielessServerHashMode.Stateless],
+            ['stateful', CookielessServerHashMode.Stateful],
+        ])('common (%s)', (_, mode) => {
+            beforeEach(async () => {
+                await setModeForTeam(mode, teamId)
+            })
+            it('should give an event a distinct id and session id ', async () => {
+                const [actual] = await cookielessServerHashStep(hub, event)
+
+                if (!actual?.properties) {
+                    throw new Error('no event or properties')
+                }
```

Comment:
I assume this is better than throwing an error

```suggestion
                expect(actual?.properties).not.toBeNull()
```

Reply:
Or maybe `not.toBeUndefined()` :)

Reply:
Or maybe `toBeDefined()`

Reply:
Yeah - it makes typescript happy, and therefore makes me happy